kamelet-support/src/main/java/io/kaoto/backend/KamelPopulator.java

```diff
@@ -266,18 +273,26 @@ public From getFlow(final List<Step> steps) {
         return from;
     }
 
-    public HashMap<String, String> buildUri(
-            final Step s,
-            final StringBuilder uri) {
+    public HashMap<String, String> buildUri(final Step s, final StringBuilder uri) {
         var params = new HashMap<String, String>();
         if (s.getParameters() != null) {
-            Collections.sort(s.getParameters());
-            for (Parameter p : s.getParameters()) {
-                if (p.isPath()) {
-                    uri.append(p.getPathSeparator());
-                    uri.append(p.getValue() != null ? p.getValue() : p.getDefaultValue());
-                } else if (p.getValue() != null) {
-                    params.put(p.getId(), p.getValue().toString());
+            //We need to make sure parameters attributes coming from the frontend are complete and right
+            //The only important thing we need to take from them is the value, the rest should be as default
+            HashMap<String, Object> values = new HashMap<>();
+            s.getParameters().stream().forEach(parameter -> values.put(parameter.getId(), parameter.getValue()));
+
+            //Now we can work with the parameters list
```

Comment:
This is where the real change is. The rest is just dragging the catalog to this point.